### ISSUE TITLE
clangd not attaching

### ISSUE THREAD

**Issue by sbeyer** [AstroNvim/AstroNvim#2285]
### Checklist

- [X] Using a stable version of Neovim (i.e. not neovim nightly)
- [X] `:AstroUpdate`
- [X] Restarted AstroNvim

### Operating system/version

Ubuntu Jammy

### Terminal/GUI

xfce4-terminal 0.8.10

### AstroNvim Health

it's fine

### Describe the bug

When I edit cpp files, `clangd` does not attach. Other LSPs work (only tested a few).

### Steps to Reproduce

1. Install clangd via mason
1. Open foo.cpp in neovim

### Expected behavior

clangd attaches

### Additional Context

Using `git revert fffc9e3` (i.e. reverting the commit "move to new mason-lspconfig syntax") solves the problem.

**Comment by mehalter** [comment]
Can you share a minimal configuration to describe what the problem is. It sounds like you have misconfigured your clangd stuff

**Comment by mehalter** [comment]
Ah, I found the issue. You need to fix your `clangd_extensions` implementation since it has had breaking changes.

**Comment by sbeyer** [comment]
Thank you so much! I didn't even get the idea that my user configuration has anything to do with it as I didn't remember that I actually configured something for C++ (except a list of treesitter parsers and LSP servers). Thanks for pointing it out. I guess I'm going to use the astrocommunity pack.

**Comment by sbeyer** [comment]
Ok I wrote the lines above before testing because they sounded reasonable.

So I checked out `main` again (without my mentioned revert) and removed my clangd_extensions configuration. Unfortunately, it did NOT resolve the problem. Using the astrocommunty pack didn't fix it neither (btw the astrocommunity pack shows that it installs clang-format every time I open a cpp file the first time after nvim started -- but this is another topic).

Disabling my whole user configuration however resolved the issue. The bad part seems to be somewhere else. I'm investigating the issue when I have more time. For now I'm back to the `git revert` solution. :/
